cpp/CMakeLists.txt

```
@@ -321,15 +321,26 @@ endif(PER_THREAD_DEFAULT_STREAM)
 
 if(BUILD_TESTS)
     include(CTest)
-    include(ConfigureGoogleTest)
+    find_package(GTest)
+    
+    if(NOT GTEST_FOUND)
+        include(ConfigureGoogleTest)
+        set(GTEST_ROOT "${CMAKE_BINARY_DIR}/googletest")
+    endif(NOT GTEST_FOUND)
+
+    add_subdirectory(${CMAKE_SOURCE_DIR}/tests)
+    
+    set(GTEST_INCLUDE_DIR "${GTEST_ROOT}/install/include")
+    set(GTEST_LIBRARY_DIR "${GTEST_ROOT}/install/lib")
+    link_directories(${GTEST_LIBRARY_DIR})
+    include_directories(${GTEST_INCLUDE_DIR})
+
+    if(NOT GTEST_FOUND)
+        set(BUILD_TESTING OFF)
+    endif(NOT GTEST_FOUND)
+
+    message(STATUS "Google C++ Testing Framework (Google Test) was found? ${GTEST_FOUND} - ${GTEST_ROOT}")
```

I don't like the potentially confusing non-grammatical "Gtest was found?" message. ("I can haz cheezburger?")

Instead, use an if / else like we had before.

```suggestion
    if(GTEST_FOUND)
        message(STATUS "Google C++ Testing Framework (Google Test) found in ${GTEST_ROOT}")
    else()
        message(AUTHOR_WARNING "Google C++ Testing Framework (Google Test) not found: automated tests are disabled.")
        set(BUILD_TESTING OFF)
    endif(NOT GTEST_FOUND)
```